### PR TITLE
Add OpenRouter provider support

### PR DESCRIPTION
**Pull request by lsm**
Adds first-class OpenRouter support using `OPENROUTER_API_KEY`, including provider registration, OpenRouter model discovery, runtime model routing, selector UX mapping, and setup docs.

Research note: Claude Code uses OpenRouter through the Anthropic-compatible base URL `https://openrouter.ai/api`, `ANTHROPIC_AUTH_TOKEN=$OPENROUTER_API_KEY`, an explicitly blank `ANTHROPIC_API_KEY`, and OpenRouter `provider/model` IDs. Compatibility is strongest with Anthropic models on OpenRouter; non-Anthropic models may not support every Claude Code feature.

Tests run locally:
- `bun run typecheck`
- `bun run lint`
- `bun run format:check`
- `cd packages/daemon && bun test --preload=./tests/unit/setup.ts ./tests/unit/1-core/providers/openrouter-provider.test.ts ./tests/unit/1-core/providers/provider-registry.test.ts ./tests/unit/1-core/core/provider-service.test.ts`
- `cd packages/web && bun run test src/hooks/__tests__/useModelSwitcher.test.ts`

CI: all required PR checks passed.

### CODE REVIEW

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Opus 4.7 (Anthropic)

> **Model:** Claude Opus 4.7 | **Client:** NeoKai | **Provider:** Anthropic

**Recommendation: REQUEST_CHANGES** — two issues need addressing before merging.

---

Solid implementation overall — the provider follows existing patterns well, tests are good, and the `fetchImpl` injection for testability is a nice improvement over sibling providers. CI is fully green. Two issues need addressing before this ships.

---

### P1 — `PROVIDER_DOT_COLORS` missing `openrouter` entry

**`packages/web/src/components/SessionStatusBar.tsx:42-48`**

The file has an explicit comment (lines 39–40): *"Keep in sync with PROVIDER_LABELS … when a new provider is added there, add a matching entry here."* But `openrouter` was not added. OpenRouter models will show a generic gray `#9CA3AF` dot instead of a branded color in the session status bar.

Add an entry, e.g.:
```ts
openrouter: { color: '#007CF0' }, // OpenRouter blue
```

### P2 — `mapRawModelsToModelInfos` defaults unknown OpenRouter models to `family = 'sonnet'`

**`packages/web/src/hooks/useModelSwitcher.ts:107-123`**

The function defaults to `family = 'sonnet'` for any model that doesn't match a known keyword. OpenRouter models like `deepseek/deepseek-r1`, `meta-ai/llama-4`, or `mistral/mistral-large` will all show the ⚡ sonnet icon, which is misleading.

The server-side `familyFromModelId` in `openrouter-provider.ts` correctly defaults unknown models to `'openrouter'`. The client should match. Add a catch-all after the keyword checks:

```ts
} else if (mid.includes('/')) {
  family = 'openrouter';
}
```

### P3 — `modelId.includes('/')` is overly broad in static fallback

**`packages/daemon/src/lib/providers/registry.ts:267`**

```ts
if (modelId === 'openrouter/auto' || modelId.includes('/')) return 'openrouter';
```

This is the static fallback for when the registry is empty (tests / early startup). The `modelId.includes('/')` would misroute any future non-OpenRouter model ID that happens to contain a `/`. Consider a more targeted check, e.g. ensuring the prefix doesn't start with `claude-`:
```ts
if (modelId === 'openrouter/auto' || (modelId.includes('/') && !modelId.startsWith('claude-'))) return 'openrouter';
```

### What looks good

- Provider implementation is complete and consistent with GLM/MiniMax patterns
- `buildSdkConfig` correctly blanks `ANTHROPIC_API_KEY` and routes through `ANTHROPIC_AUTH_TOKEN` + `ANTHROPIC_BASE_URL`
- `delete` → `= ''` for `ANTHROPIC_API_KEY` in `ProviderService` is safe — `restoreEnvVars` handles both `undefined` and value restores correctly
- `isProbablyOpenRouterKey` heuristic is reasonable (`sk-or-` prefix is stable per OpenRouter docs)
- `fetchImpl` constructor injection makes `getModels()` fully testable — nice pattern
- Test coverage covers identity, availability, SDK config, tier mapping, model fetching, and auth error handling
- Docs are clear and include compatibility caveats
- Network allowlist correctly includes `openrouter.ai` / `*.openrouter.ai`

File: packages/daemon/src/lib/providers/registry.ts [hunk 1]
```diff
@@ -263,6 +264,7 @@ export function inferProviderForModel(modelId: string): ProviderIdStr {
 	// Static fallback when registry is empty
 	if (modelId.startsWith('glm-') || modelId === 'glm') return 'glm';
 	if (modelId.startsWith('minimax-') || modelId === 'minimax') return 'minimax';
+	if (modelId === 'openrouter/auto' || modelId.includes('/')) return 'openrouter';
```
Comment (lsm) [inline packages/daemon/src/lib/providers/registry.ts]: **P3** — `modelId.includes('/')` is overly broad for the static fallback path. While no existing model IDs contain `/`, a future non-OpenRouter model could. Consider a more targeted check: `modelId.includes('/') && !modelId.startsWith('claude-')`.

File: packages/web/src/hooks/useModelSwitcher.ts [hunk 2]
```diff
@@ -111,9 +114,11 @@ export function mapRawModelsToModelInfos(models: RawModelEntry[]): ModelInfo[] {
 			family = 'glm';
 		} else if (mid.startsWith('minimax-')) {
 			family = 'minimax';
-		} else if (mid.startsWith('gpt-')) {
+		} else if (mid === 'openrouter/auto') {
```
Comment (lsm) [inline packages/web/src/hooks/useModelSwitcher.ts:117]: **P2** — This only catches `openrouter/auto`. Other OpenRouter models like `deepseek/deepseek-r1` or `meta-ai/llama-4` will fall through to the default `family = 'sonnet'` and show the ⚡ sonnet icon. Add a broader catch-all after all keyword checks: `else if (mid.includes('/')) { family = 'openrouter'; }` — this matches the server-side `familyFromModelId` default in `openrouter-provider.ts`.

File: packages/daemon/src/lib/providers/registry.ts (hunk 1 above)
Comment (lsm) [reply packages/daemon/src/lib/providers/registry.ts]: Fixed in 7dedd0e: the static fallback now only treats slash IDs as OpenRouter when they do not start with claude-, and provider-registry tests cover both deepseek/deepseek-r1 and a claude-* slash edge case.

File: packages/web/src/hooks/useModelSwitcher.ts (hunk 2 above)
Comment (lsm) [reply packages/web/src/hooks/useModelSwitcher.ts:117]: Fixed in 7dedd0e: unknown slash-based OpenRouter model IDs now fall through to family openrouter after the GPT/Gemini keyword checks, and the model switcher test covers deepseek/deepseek-r1.

**Comment by lsm** [comment]
Addressed the remaining review-body item in 7dedd0e as well: SessionStatusBar now has an OpenRouter provider dot color (#007CF0).

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Opus 4.7 (Anthropic)

> **Model:** Claude Opus 4.7 | **Client:** NeoKai | **Provider:** Anthropic

### Verdict: APPROVE

All three findings from the previous review have been addressed correctly:

- **P1** ✅ `PROVIDER_DOT_COLORS` now includes `openrouter: { color: '#007CF0' }` in `SessionStatusBar.tsx`.
- **P2** ✅ `mapRawModelsToModelInfos` now has a `mid.includes('/')` catch-all that maps unrecognized OpenRouter models to `family = 'openrouter'`. Test covers `deepseek/deepseek-r1` → `openrouter`.
- **P3** ✅ `inferProviderForModel` static fallback now uses `modelId.includes('/') && !modelId.startsWith('claude-')`. Test covers `claude-sonnet-4.6/preview` → `anthropic`.

CI is fully green. LGTM — ready to merge.
